### DISCUSSION TITLE
Request to add Bitwise Operations

### DISCUSSION THREAD

**Discussion by Calemy** [meilisearch/product#445] [Feedback & Feature Proposal]
It would be very advantageous to have bitwise operations on filtering like mysql or mongodb. It would give a lot of new options to store data. I personally use it myself and would really like to see that option being added to meilisearch. Didn't see any thread mentioning this before.

**Comment by gmourier** [comment]
Hi @Mxnuuel 👋

Thanks for your feedback.

Can you share with us a use case and one/two examples of using bitwise operations? It will be greatly appreciated!

Thanks

**Reply by Calemy** [reply]
Following code is from a personal project of mine

Here the enum created depending on the modes (0-3) and gets safed in the database
![crawler.js](https://x.catboy.best/pTNqLf8.png)

Here i would retrieve the data via the Bitoperator to check if the mode given over the search url fits one result.
![search.js](https://x.catboy.best/SCmlTX0.png)

Hope that showes my personal usecase well enough!
Would make my life a lot easier than to store the array since i want to add multiple mode search support.

**Reply by gmourier** [reply]
Thanks, @Mxnuuel 👋

This is not a priority for Meilisearch at this time. 

The discussion remains open so other users could express their needs for the bitwise operator.
